Mathlib/Topology/UniformSpace/Path.lean

```diff
@@ -0,0 +1,69 @@
+/-
+Copyright (c) 2025 Yury Kudryashov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yury Kudryashov
+-/
+import Mathlib.Topology.Path
+import Mathlib.Topology.UniformSpace.CompactConvergence
+import Mathlib.Topology.UniformSpace.HeineCantor
+import Mathlib.Topology.MetricSpace.Lipschitz
+import Mathlib.Topology.ContinuousMap.Interval
+
+/-!
+# Paths in uniform spaces
+
+In this file we define a `UniformSpace` structure on `Path`s
+between two points in a uniformspace
+and prove that various functions associated with `Path`s are uniformly continuous.
+-/
+
+open scoped unitInterval Topology Uniformity
+
+variable {X : Type*} [UniformSpace X] {x y z : X}
+
+namespace Path
+
+instance instUniformSpace : UniformSpace (Path x y) :=
+  .comap ((↑) : _ → C(I, X)) ContinuousMap.compactConvergenceUniformSpace
+
+theorem isUniformEmbedding_coe : IsUniformEmbedding ((↑) : Path x y → C(I, X)) where
+  comap_uniformity := rfl
+  injective := ContinuousMap.coe_injective'
+
+theorem uniformContinuous (γ : Path x y) : UniformContinuous γ :=
+  CompactSpace.uniformContinuous_of_continuous <| map_continuous _
+
+/-- Extension of a path to the real line is a uniformly continuous function. -/
+theorem uniformContinuous_extend (γ : Path x y) : UniformContinuous γ.extend :=
+  γ.uniformContinuous.comp <| LipschitzWith.projIcc _ |>.uniformContinuous
+
+/-- Extension of a path to the real line is uniformly continuous in the path argument. -/
+theorem uniformContinuous_extend_left : UniformContinuous (Path.extend : Path x y → _) :=
+  have : Fact ((0 : ℝ) ≤ 1) := ⟨zero_le_one⟩
```

Comment:
Hasn't this been registered as a global instance in current master? It's worth merging master and checking if this can be removed.